### ISSUE TITLE
There seems to be something wrong with 0.17 when using subscriptions

### ISSUE THREAD

**Issue by danhawkins** [elixir-ecto/postgrex#649]
### Elixir version

1.14.4

### Database and Version

postgres 15

### Postgrex Version

0.17

### Current behavior

I am using https://github.com/commanded/eventstore, when using version 0.17 I get no subscription updates

### Expected behavior

I expect to get updates from subscriptions while using https://github.com/commanded/eventstore, I verified that rolling back to 16.5 works

**Comment by josevalim** [comment]
We have no familiarity with commando, so can you please report it to them and maybe they can isolate it? Or at least provide a failing test case? Thank you.

**Comment by jkmrto** [comment]
We have a similar error with version **0.17.0**, also using [Commanded](https://github.com/commanded/eventstore). It also works when using version **0.16.5** of **postgrex**.

We get this (debug) log in local: 

```
[debug] Attempting to start Postgrex.Notifications
[debug] Successfully started Postgrex.Notifications (#PID<0.3135.0>)
[debug] Monitored process EXIT due to: {:function_clause, [{Postgrex.SimpleConnection, :handle_event, [:timeout, nil, :no_state
, %Postgrex.SimpleConnection{idle_interval: 5000, protocol: %Postgrex.Protocol{sock: {:gen_tcp, #Port<0.47>}, connection_id: 31
0, connection_key: -674320564, peer: {{127, 0, 0, 1}, 5432}, types: {EventStore.PostgresTypes, #Reference<0.4085729420.21208760
33.141520>}, null: nil, timeout: 15000, ping_timeout: 15000, parameters: #Reference<0.4085729420.2120744965.137402>, queries: #
Reference<0.4085729420.2120876033.146109>, postgres: :idle, transactions: :naive, buffer: :active_once, disconnect_on_error_cod
es: [], scram: nil}, auto_reconnect: false, reconnect_backoff: 500, state: {Postgrex.Notifications, %Postgrex.Notifications{fro
m: nil, ref: nil, auto_reconnect: false, connected: true, listeners: %{#Reference<0.4085729420.2120744961.146114> => {"events",
 #PID<0.2514.0>}}, listener_channels: %{"events" => %{#Reference<0.4085729420.2120744961.146114> => #PID<0.2514.0>}}}}}], [file
: 'lib/postgrex/simple_connection.ex', line: 340]}, {:gen_statem, :loop_state_callback, 11, [file: 'gen_statem.erl', line: 1428
]}, {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 240]}]}
```

**Comment by josevalim** [comment]
Can you please try the just released v0.17.1?

**Comment by danhawkins** [comment]
Thank @josevalim 🙏
